controller/pokemon.go

```diff
@@ -0,0 +1,60 @@
+package controller
+
+import (
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	"github.com/rmonroy-wiz/ondemand-go-bootcamp-2022/business"
+)
+
+type pokemon struct {
+	pokemonBusiness business.PokemonBusiness
+}
+
+type Pokemon interface {
+	GetAllPokemons()
+}
+
+func NewPokemonController(pokemonBusiness business.PokemonBusiness) *pokemon {
+	return &pokemon{
+		pokemonBusiness: pokemonBusiness,
+	}
+}
+func (p pokemon) GetAllPokemons(c *gin.Context) {
+	pokemons, err := p.pokemonBusiness.GetAll()
+
+	if err != nil {
+		c.AbortWithStatus(http.StatusNotFound)
+	}
+
+	c.JSON(http.StatusOK, pokemons)
+}
+
+// GetPokemonByID get pokemon based on ID
+func (p pokemon) GetPokemonByID(c *gin.Context) {
+	id, err := strconv.Atoi(c.Params.ByName("pokemonId"))
+	if err != nil {
+		c.AbortWithStatus(http.StatusNotFound)
+	}
+	pokemon, err := p.pokemonBusiness.GetByID(id)
+	if err != nil {
```

Comment:
Is a good practice to log the errors in the uppermost layer or where you are handling them and returning something else (recovering from the error and returning a value or returning a different error)

can you please log the error here?